src/main/java/seedu/address/model/event/fields/DateTime.java

```diff
@@ -0,0 +1,39 @@
+package seedu.address.model.event.fields;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents an Event's phone number in the Calendar.
+ * Guarantees: immutable; is valid as declared in {@link #isValidDateTime(String)}
+ */
+public class DateTime {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Date and Time must be specified! "
+                    + "The following format must be adhered: YYYY-MM-DD HHMM e.g. 2023-03-23 1200!";
+    private static DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm");
+
+    public final LocalDateTime dateTime;
+
+    public DateTime(String dateTime) {
+        this.dateTime = LocalDateTime.parse(dateTime, dateTimeFormatter);
+    }
+
+    public LocalDateTime getDateTime() {
+        return this.dateTime;
+    }
+
+    /**
+     * Returns true if a given string is a valid Date Time.
+     */
+    public static boolean isValidDateTime(String duration) {
```

Comment:
This function would have to change if we want different formatters. Just to note.